msdsl/model.py

```diff
@@ -504,16 +515,37 @@ def set_from_sync_func(self, signal: Union[Signal, str], func: GeneralFunction,
         addr_frac = self.set_this_cycle(self.get_next_name(f'{func.name}_addr_frac_'), addr_frac_expr,
                                         range_=1.01)
 
-        # look up coefficient values
-        coeffs = [self.get_next_name(f'{func.name}_coeff_{k}_') for k in range(func.order+1)]
-        for k, coeff in enumerate(coeffs):
-            if isinstance(func, PlaceholderFunction):
-                self.set_from_sync_ram(signal=coeff, format_=func.formats[k], addr=addr_mux,
-                                       clk=clk, ce=ce, we=we, din=wdata[k])
-            elif isinstance(func, Function):
-                self.set_from_sync_rom(signal=coeff, table=func.tables[k], addr=addr_uint, clk=clk, ce=ce)
-            else:
-                raise Exception(f'Unknown function type: {func}')
+        # Create a list of functions to be implemented from the same input.
+        # For Function and PlaceholderFunction, that list will only have one entry,
+        # but this makes it possible to share code between the single- and multi-
+        # output function versions.
+        if isinstance(func, MultiFunction):
```

Comment:
functions can now have multiple outputs (but the single output case works just as before)